source/common/http/context_impl.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "envoy/http/context.h"
+
+#include "common/http/codes.h"
+
+namespace Envoy {
+namespace Http {
+
+/**
+ * Captures http-related structures with cardinality of one per server.
+ */
+class ContextImpl : public Context {
+public:
+  ContextImpl();
+  virtual ~ContextImpl();
+
+  Tracing::HttpTracer& tracer() override { return *tracer_; }
+  CodeStats& codeStats() override { return code_stats_; }
+
+  void setTracer(Tracing::HttpTracer& tracer) { tracer_ = &tracer; }
+
+private:
+  Tracing::HttpTracerPtr tracer_storage_;
```

Comment:
Is this actually used? If you need it for a default null tracer can you just store a null tracer directly as a member variable?